api/cmd/api/main.go

```diff
@@ -193,6 +190,21 @@ func main() {
 		fmt.Fprintf(os.Stderr, "invalid host argument: %q (valid hosts: localhost)\n", *hostF)
 	}
 
+	r := mux.NewRouter()
+
+	authPort := "4200"
+
+	auth.AuthProvider(r, api)
+	go func() {
+		// start the web server on port and accept requests
+		logger.Infof("AUTH server listening on port %q", authPort)
+		logger.Fatal(http.ListenAndServe(":"+authPort,
```

Comment:
what would happen if the auth server fail due to some Fatal error?
would it stop both the server?

Reply:
Ideally it should not stop both the server

Reply:
because both are running on different port

Reply:
then how would one know that auth server is failing?

Reply:
if pod is running even if auth server is failing

Reply:
and would it be same for the other? if api server fails will auth server still work?

Reply:
yes if auth server returns error at some point then the api server would still be in running state.
```
2021-11-15T20:44:36.247+0530	INFO	log/logger.go:33	HTTP Request	{"service": "http", "id": "aq_UyPLp", "status": 200, "bytes": 611, "time": "1.071069ms"}
2021-11-15T20:44:36.247+0530	ERROR	service/auth.go:143	hello world error	{"service": "auth"}
github.com/tektoncd/hub/api/pkg/auth/service.(*service).HubAuthenticate
	/home/vj/go/src/github.com/tektoncd/hub/api/pkg/auth/service/auth.go:143
net/http.HandlerFunc.ServeHTTP
	/usr/local/go/src/net/http/server.go:2049
github.com/gorilla/mux.(*Router).ServeHTTP
	/home/vj/go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210
github.com/gorilla/handlers.(*cors).ServeHTTP
	/home/vj/go/pkg/mod/github.com/gorilla/handlers@v1.5.1/cors.go:138
net/http.serverHandler.ServeHTTP
	/usr/local/go/src/net/http/server.go:2867
net/http.(*conn).serve
	/usr/local/go/src/net/http/server.go:1932
2021-11-15T20:44:36.248+0530	INFO	log/logger.go:33	HTTP Request	{"service": "http", "id": "FCXdNplQ", "req": "GET /v1/catalogs", "from": "::1"}
2021-11-15T20:44:36.248+0530	INFO	log/logger.go:33	HTTP Request	{"service": "http", "id": "kYJJ0u5Q", "req": "GET /resources", "from": "::1"}
2021-11-15T20:44:36.249+0530	INFO	app/gorm_logger.go:85	/home/vj/go/src/github.com/tektoncd/hub/api/v1/service/catalog/catalog.go:46
[1.664ms] [rows:1] SELECT * FROM "catalogs" WHERE "catalogs"."deleted_at" IS NULL ORDER BY id	{"service": "catalog", "id": "FCXdNplQ"}

```
If you see the above case I manually logged the error randomly in one of the auth functions, the API server still worked

Reply:
should we fix this 🤔 
I am not completely sure !
we can use a channel and notify on it 🤔

Reply:
lets keep this as it is for now .. and think a bit on this
do we have a status api for auth?

Reply:
I think we have a status api